collection/network/get-mac-address.yml

```diff
@@ -2,21 +2,27 @@ rule:
   meta:
     name: get MAC address
     namespace: collection/network
-    author: moritz.raabe@fireeye.com
+    author:
+      - moritz.raabe@fireeye.com
+      - joakim@intezer.com
     scope: function
     att&ck:
       - Discovery::System Information Discovery [T1082]
     examples:
       - al-khaser_x64.exe_:0x14001A1BC
   features:
-    - and:
-      - api: iphlpapi.GetAdaptersInfo
-      - or:
-        - offset: 0x194 = IP_ADAPTER_INFO.Address
-        - offset: 0x195 = IP_ADAPTER_INFO.Address+1
-        - offset: 0x196 = IP_ADAPTER_INFO.Address+2
-        - offset: 0x197 = IP_ADAPTER_INFO.Address+3
-        - offset: 0x198 = IP_ADAPTER_INFO.Address+4
-        - offset: 0x199 = IP_ADAPTER_INFO.Address+5
-      - optional:
-        - string: "%02X-%02X-%02X-%02X-%02X-%02X"
+    - or:
+      - and:
+        - api: iphlpapi.GetAdaptersInfo
+        - or:
+          - offset: 0x194 = IP_ADAPTER_INFO.Address
+          - offset: 0x195 = IP_ADAPTER_INFO.Address+1
+          - offset: 0x196 = IP_ADAPTER_INFO.Address+2
+          - offset: 0x197 = IP_ADAPTER_INFO.Address+3
+          - offset: 0x198 = IP_ADAPTER_INFO.Address+4
+          - offset: 0x199 = IP_ADAPTER_INFO.Address+5
+        - optional:
+          - string: "%02X-%02X-%02X-%02X-%02X-%02X"
+      - and:
+        - match: read file
+        - string: /\/sys\/class\/net\/\S+\/address/
```

Comment:
yuck these paths are ugly.

i wonder if substring literals are common enough for us to introduce a new feature/format. perhaps something like:

  - `substring: "/etc/passwd"`
  - `string/substring: "/etc/passwd"`
  - `string: "/etc/passwd"g

note: this specific path isn't the best example because its *actually* a regex patter, but you get the idea.

Reply:
It may look ugly because it needs a wildcard for the name of the network adapter. For example `/sys/class/net/lo/address`, `/sys/class/net/docker0/address`.